### DISCUSSION TITLE
Documentation for Klipper

### DISCUSSION THREAD

**Discussion by keymaster777** [feelfreelinux/octo4a#42] [Ideas]
Hey I was wondering if after the klipper extension has been perfected you could add some resources or a guide for the set-up including flashing with a separate machine. I love the app and I'm completely baffled that you've made it so easy to have octoprint and klipper on an android without even needing it to be rooted. Thanks for your hard work, if you want a guinea pig for working through the klipper documentation I am happy to help, until then I'll keep using Marlin.

**Comment by feelfreelinux** [comment]
Initial documentation for Klipper provided in: https://github.com/feelfreelinux/octo4a/discussions/59

Would love to hear what you think about it!

**Reply by yanggao1898** [reply]
Hello! I've followed the linked documentation, and got a little bit farther along with setting up Octo4a and Klipper than before! Now i'm getting some errors about mcu in klippy.log... I seem to be able to connect to the printer, but when i type Status, i get this:
```
Send: STATUS
Recv: // Printer is not ready
Recv: // The klippy host software is attempting to connect.  Please
Recv: // retry in a few moments.
Recv: // Klipper state: Not ready
Recv: !! Printer is not ready
Changing monitoring state from "Operational" to "Error"
Send: M112
Send: N2 M112*35
Send: N3 M104 T0 S0*34
Send: N4 M140 S0*97
Changing monitoring state from "Error" to "Offline after error"
Connection closed, closing down monitor
```

And in klippy.log i get this (repeated over and over): 
```
mcu 'mcu': Timeout on connect
mcu 'mcu': Wait for identify_response
Traceback (most recent call last):
  File "/klipper/klippy/serialhdl.py", line 69, in _get_identify_data
    params = self.send_with_response(msg, 'identify_response')
  File "/klipper/klippy/serialhdl.py", line 258, in send_with_response
    return src.get_response([cmd], self.default_cmd_queue)
  File "/klipper/klippy/serialhdl.py", line 315, in get_response
    self.serial.raw_send_wait_ack(cmds[-1], minclock, reqclock,
  File "/klipper/klippy/serialhdl.py", line 250, in raw_send_wait_ack
    self._error("Serial connection closed")
  File "/klipper/klippy/serialhdl.py", line 62, in _error
    raise error(self.warn_prefix + (msg % params))
serialhdl.error: mcu 'mcu': Serial connection closed

```

**Reply by JoshBlackwood** [reply]
I'm getting the same errors after following the guide, and my first guess after a few minutes poking around in SSH is that my unrooted Amazon Fire tablet is blocking permissions to /dev, as the default ssh user can't even ls /dev. I'll try it with a rooted Nexus 5 on a more standard OS and report back.

**Reply by JoshBlackwood** [reply]
> I'm getting the same errors after following the guide, and my first guess after a few minutes poking around in SSH is that my unrooted Amazon Fire tablet is blocking permissions to /dev, as the default ssh user can't even ls /dev. I'll try it with a rooted Nexus 5 on a more standard OS and report back.

I had not paid enough attention to the app itself, and hadn't selected a driver upon connecting the printer. Once I found the right driver and chose it, Octoprint worked as intended with no more errors in the log.

**Reply by GreenG20t** [reply]
Josh, we're you able to get it to run on the fire tab? What version? I have a few laying around, but I assumed they would not have enough power.

**Reply by solarkraft** [reply]
> selected a driver upon connecting the printer

Sorry, what does this mean? Neither Octo4a, nor OctoKlipper, nor Klipper seem to have the concept of a "driver".

**Reply by solarkraft** [reply]
nvm, it's a setting for the USB Serial connection. Sadly I can't change it on mine ...

**Reply by JoshBlackwood** [reply]
I don't have it running anymore, unfortunately, I never did get an adapter so I could have it connected and charging at the same time, so couldn't leave it for long prints. If I recall correctly it was a selection that appears in the app on connecting the USB with the app open, it lets you choose the serial connection driver. It was trial and error, but the second or third option worked for me.
